File: src-architecture/game/GameLoader.java
package game;

import engine.thread.Tickable;
import engine.utils.MeshLoader;
import game.architecture.GameComponentContainer;
import game.creator.ApolloSystemCreator;
import game.creator.SolarSystemCreator;
import model.ship.Mesh;
import model.ship.PlayerShip;
import model.system.StarSystem;

import javax.media.opengl.GL2;
import java.io.IOException;
import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;

public class GameLoader extends GameComponentContainer implements Tickable {

    private List<StarSystem> starSystems;
    private PlayerShip playerShip;

    @Override
    public void tick() {
        GameComponentContainer.starSystemManager.tick(playerShip);
    }

    public void loadSystem(PlayerShip playerShip) {

        System.out.println("Initializing...");
        starSystems = new LinkedList<>();
        StarSystem solarSystem = null;
        StarSystem apolloSystem;

        List<Mesh> meshes = new ArrayList<Mesh>();

        System.out.println("Loading Meshes...");

        // meshes.add(new MeshDrawer(gl, MeshLoader.loadFromFile("wraith", "wraith")));
        // meshes.add(MeshLoader.loadFromFile("exxon", "Alicia"));
        meshes.add(MeshLoader.loadFromFile("alien", "alien-mothership-green"));

        meshRepository.add(meshes);

        GameComponentContainer.meshRepository.add(meshes);

        try {
            System.out.println("Loading Solar system...");
            solarSystem = new SolarSystemCreator().create();
            solarSystem.setVisible(true);

            starSystems.add(solarSystem);
            celestialBodyRepository.add(solarSystem.getCelestialBodies());
            starSystemRepository.add(solarSystem);

            System.out.println("Loading Apollo system...");
            apolloSystem = new ApolloSystemCreator().create();
            apolloSystem.setVisible(false);

            starSystems.add(apolloSystem);
            celestialBodyRepository.add(apolloSystem.getCelestialBodies());
            starSystemRepository.add(apolloSystem);
        } catch (IOException e) {
            e.printStackTrace();
        }

        this.playerShip = playerShip;
        this.playerShip.setStarSystem(solarSystem);

        System.out.println("Done!");
    }

    public void drawCurrentSkyBox(GL2 gl, PlayerShip playerShip) {
        StarSystem system = playerShip.getStarSystem();
        system.getSkyBox().draw(gl);
    }

    public void drawObjects(GL2 gl, PlayerShip playerShip) {
        GameComponentContainer.starSystemManager.draw(gl, playerShip);
    }
}

File: src-architecture/game/GameLoader.kt
package game

import engine.thread.Tickable
import engine.utils.MeshLoader
import game.architecture.GameComponentContainer
import game.creator.ApolloSystemCreator
import game.creator.SolarSystemCreator
import model.ship.Mesh
import model.ship.PlayerShip
import model.system.StarSystem
import java.util.*
import javax.media.opengl.GL2

class GameLoader(
        private var playerShip: PlayerShip,
        private var starSystems: MutableList<StarSystem> = mutableListOf()
) : GameComponentContainer(), Tickable {

    override fun tick() {
        val system = playerShip.starSystem

        for (celestialBody in system!!.celestialBodies) {
            celestialBody.tick(playerShip)
        }

        GameComponentContainer.shipManager.tick(system.ships, playerShip)
    }

    fun loadSystem(playerShip: PlayerShip) {
        println("Initializing...")

        println("Loading Meshes...")
        val meshes = ArrayList<Mesh>()
        meshes.add(MeshLoader.loadFromFile("exxon", "alien-mothership-green"))
        meshes.add(MeshLoader.loadFromFile("alien", "alien-mothership-green"))
        GameComponentContainer.meshRepository.add(meshes)
        GameComponentContainer.meshRepository.add(meshes)

        println("Loading Solar system...")
        val solarSystem: StarSystem = SolarSystemCreator().create()
        starSystems.add(solarSystem)

        GameComponentContainer.celestialBodyRepository.add(solarSystem.celestialBodies)
        GameComponentContainer.starSystemRepository.add(solarSystem)

        println("Loading Apollo system...")
        val apolloSystem: StarSystem = ApolloSystemCreator().create()
        starSystems.add(apolloSystem)

        GameComponentContainer.celestialBodyRepository.add(apolloSystem.celestialBodies)
        GameComponentContainer.starSystemRepository.add(apolloSystem)

        this.playerShip.starSystem = solarSystem

        println("Done!")
    }

    fun drawCurrentSkyBox(gl: GL2, playerShip: PlayerShip) {
        val system = playerShip.starSystem
        system!!.skyBox.draw(gl)
    }

    fun drawObjects(gl: GL2, playerShip: PlayerShip) {
        val system = playerShip.starSystem

        for (body in system!!.celestialBodies) {
            body.draw(gl)
        }

        GameComponentContainer.shipManager.draw(gl, system.ships)
    }
}
